### PR TITLE
Fix module selectors in mypy's module overrides

### LINKED ISSUE #12238: Many hidden type errors due to wrong mypy config
In `pyproject.toml` in `[[tool.mypy.overrides]]` section we use module selectors like `"bokeh.command"`, but that only matches the named module, but not any of its submodules. The correct syntax is `"bokeh.command.*"`.

### PR DESCRIPTION
**Pull request by mattpap**
This still requires fixing all the type errors.

fixes #12238

### CODE REVIEW

**Comment by bryevdv** [comment]
I think, annoyingly, that both forms are needed. Some of the `.*` are missing not in error, but because the work on submodules was still tbd.

File: pyproject.toml [hunk 1]
```diff
@@ -68,21 +70,21 @@ module = [
     "bokeh.driving",
     "bokeh.embed.standalone",
     "bokeh.embed.wrappers",
-    "bokeh.ext",
-    "bokeh.io",
+    "bokeh.ext.*",
+    "bokeh.io", # TODO: .*;  mypy_path resolution is broken
```
Comment (mattpap) [inline pyproject.toml:74]: mypy is completely broken right now, when it comes to local `typings/` resolution. It's fixed, but we will need to wait for a release.

File: bokeh/io/notebook.py [hunk 2]
```diff
@@ -628,14 +628,16 @@ def _origin_url(url: str) -> str:
...
 
     '''
+    port_ = f":{port}" if port is not None else ""
+
     if url.startswith("http"):
-        return '%s:%d%s' % (url.rsplit(':', 1)[0], port, "/")
+        return '%s%s%s' % (url.rsplit(':', 1)[0], port_, "/")
     else:
-        return 'http://%s:%d%s' % (url.split(':')[0], port, "/")
+        return 'http://%s%s%s' % (url.split(':')[0], port_, "/")
```
Comment (bryevdv) [inline bokeh/io/notebook.py:640]: as long as these are being modified why not also update to f-strings

File: bokeh/sampledata/stocks.py [hunk 3]
```diff
@@ -67,20 +68,29 @@
 # Private API
 #-----------------------------------------------------------------------------
 
-def _read_data(name):
+class StocksData(TypedDict):
```
Comment (bryevdv) [inline bokeh/sampledata/stocks.py:71]: This structure holds info for a single stock so `StockData` (or maybe `TickerData`) would be better I think

File: pyproject.toml [hunk 4]
```diff
@@ -68,21 +70,21 @@ module = [
...
     "bokeh.palettes",
     "bokeh.plotting._tools",
-    "bokeh.protocol",
-    "bokeh.sampledata",
+    "bokeh.protocol.*",
```
Comment (bryevdv) [inline pyproject.toml:77]: Please do conduct a test to make sure that `bokeh.protocol` is not needed in addition to `bokeh.protocol.*` etc. My recollection is that the latter only covers submodules an not the package itself

File: pyproject.toml (hunk 1 above)
Comment (mattpap) [reply pyproject.toml:74]: mypy 0.971 is out, so hopefully this will be resolved soon.

File: pyproject.toml (hunk 4 above)
Comment (mattpap) [reply pyproject.toml:77]: I tested and `.*` covers everything.

**Pull request merged**